javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -334,11 +348,7 @@ default int leafCount() {
      * @return The number of nodes of this tree.
      */
     default int nodeCount() {
-        if (isEmpty()) {
-            return 0;
-        } else {
-            return 1 + getChildren().foldLeft(0, (count, child) -> count + child.nodeCount());
-        }
+        return length();
```

Comment:
@danieldietrich I am not sure if you get notifications when I force push an updated commit. I applied your last suggestions